mkdocs/utils/compat.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import unicode_literals
+
+import os
+import sys
+
+try:                                                          # pragma: no cover
+    from urllib.request import pathname2url as _pathname2url  # noqa
+except ImportError:                                           # pragma: no cover
+    from urllib import pathname2url as _pathname2url          # noqa
+
+
+def pathname2url(p):
+    """This is a backport from Python 3.5, it will only be used on Python 2
+    under windows.
+
+    # TODO: Remove this function with Python 2 support is dropped
+    """
+
+    # If we are on Python 3 already, or not using Windows then use the version
+    # that shipped with that Python. 3.3 is the lowest Python 3 version we
+    # support.
+    if sys.version_info > (3, 3) or os.name != 'nt':
+        return _pathname2url(p)
+
+    # The rest of this function is a direct copy from cpython/Lib/nturl2path.py
+    # with only a few PEP8(!) fixes.
+
+    """OS-specific conversion from a file system path to a relative URL
+    of the 'file' scheme; not recommended for general use."""
+    # e.g.
+    #   C:\foo\bar\spam.foo
+    # becomes
+    #   ///C:/foo/bar/spam.foo
+    import urllib.parse
```

Comment:
Maybe just do `from urllib import quote` as `quote` is the only thing actually used.